scripts/speech_recognition/convert_to_tarred_audio_dataset.py

```diff
@@ -644,17 +568,28 @@ def _read_manifest(self, manifest_path: str, config: ASRTarredDatasetConfig):
 
         return entries, total_duration, filtered_entries, filtered_duration
 
-    def _write_to_tar(self, tar, audio_filepath: str, squashed_filename: str) -> None:
-        if (codec := self.config.force_codec) is None or audio_filepath.endswith(f".{codec}"):
+    def _write_to_tar(
+        self, tar, audio_filepath: str, squashed_filename: str, duration: float = None, offset: float = 0
+    ) -> None:
+        if ((codec := self.config.force_codec) is None or audio_filepath.endswith(f".{codec}")) and not duration:
```

Comment:
There are so many if/else here now that I think we should have some unit test coverage for this script (or just the function).